test/test_mutable_binheap.jl

```diff
@@ -157,6 +157,16 @@ end
         @test pop!(h) == 2
         @test isequal(heap_values(h), [7, 10])
         @test isequal(list_values(h), [10, 7])
+        empty!(h.nodes)
```

Comment:
This should be its own separate testset.
And rather than `empty!(h.nodes)` it should just declare a new heap `h`.


Run-on tests become increasingly hard to read/edit.
And we have too many of them as it is.